From 51aed24f9d98fd2985b4383dbe6ca7138df5df90 Mon Sep 17 00:00:00 2001
From: Richard Gong <richard.gong@intel.com>
Date: Mon, 10 Aug 2020 14:33:40 -0500
Subject: [PATCH 105/151] arm64: dts: socfpga_dm: correct incorrect definitions
 for watchdog

commit  784a45f24ad6f27029612ad169bdfcbdfe63e6dd from
https://github.com/altera-opensource/linux-socfpga.git
branch is socfpga-5.4.64-lts

Simics team reported the watchdog reboots the system, they made some
updates at dtsi file to resolve watchdog reboot issue.

Port the updates to LTS mainline kernel.

Signed-off-by: Richard Gong <richard.gong@intel.com>
Signed-off-by: Markov, IgorX <igorX.markov@intel.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
Integrated-by: Jun Zhang <jun.zhang@windriver.com>
---
 arch/arm64/boot/dts/intel/socfpga_dm.dtsi | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/intel/socfpga_dm.dtsi b/arch/arm64/boot/dts/intel/socfpga_dm.dtsi
index c2fda1ff7678..b998a8285ad5 100644
--- a/arch/arm64/boot/dts/intel/socfpga_dm.dtsi
+++ b/arch/arm64/boot/dts/intel/socfpga_dm.dtsi
@@ -96,6 +96,13 @@
 		ranges = <0 0 0 0xffffffff>;
 
 /* Use fixed frequency clocks in Simics - remove clk manager for initial DT */
+		clocks {
+			wdt_clk: wdt-clk {
+				#clock-cells = <0>;
+				compatible = "fixed-clock";
+				clock-frequency = <100000000>;
+			};
+		};
 
 		gmac0: ethernet@ff800000 {
 			compatible = "altr,socfpga-stmmac-a10-s10", "snps,dwmac-3.74a", "snps,dwmac";
@@ -417,7 +424,7 @@
 			reg = <0xffd00200 0x100>;
 			interrupts = <0 117 4>;
 			resets = <&rst WATCHDOG0_RESET>;
-			clock-frequency = <100000000>;
+			clock-frequency = <&wdt_clk>;
 			status = "disabled";
 		};
 
@@ -426,7 +433,7 @@
 			reg = <0xffd00300 0x100>;
 			interrupts = <0 118 4>;
 			resets = <&rst WATCHDOG1_RESET>;
-			clock-frequency = <100000000>;
+			clock-frequency = <&wdt_clk>;
 			status = "disabled";
 		};
 
@@ -435,7 +442,7 @@
 			reg = <0xffd00400 0x100>;
 			interrupts = <0 125 4>;
 			resets = <&rst WATCHDOG2_RESET>;
-			clock-frequency = <100000000>;
+			clock-frequency = <&wdt_clk>;
 			status = "disabled";
 		};
 
@@ -444,7 +451,7 @@
 			reg = <0xffd00500 0x100>;
 			interrupts = <0 126 4>;
 			resets = <&rst WATCHDOG3_RESET>;
-			clock-frequency = <100000000>;
+			clock-frequency = <&wdt_clk>;
 			status = "disabled";
 		};
 
-- 
2.26.1

